EIPS/eip-X-ASIC-resistant-block-mining.md

```diff
@@ -0,0 +1,180 @@
+
+A new field of `EthashVersion` defined as an 8 bit unsigned enumeration is added to 
+the the block header. If this field is absent, its value is assumed to be equal to
+ zero.
+
+The enumeration should be forward compatible and defined as:
+`EthashUndefined = 0x00  
+EthashVersion1 = 0x01  
+EthashVersion2 = 0x02  
+...  
+EthashVersion255 = 0xFF`
+
+When this field is present and set to 0x02, the `mine` and `VerifySeal` operations
+utilize the `ethashV2` algorithm. If this field is set to any value other than 0x01,
+0x02, or 0x00 `VerifySeal` shall reject the block.
+
+`VerifySeal` shall also fail verification in the event 
+`block.Number >= ASIC_MITIGATION_FORK_BLKNUM && block.EthashVersion < EthashVersion2`
+
+## Agent Changes
+ 
+GetWork may optionally return the proposed blocks `EthashVersion` field. While a 
+miner or pool may infer the requirement for ethashV2 based on the computed 
+epoch of the provided seedHash, it may be beneficial to explicitly provide this
+field so a miner does not require special configuration when mining on a chain
+that chooses not to implement the ASIC_Mitigation hardfork.
+
+Due to compatibility concerns with implementations that already add additional 
+parameters to GetWork, it may be desired to define a 33 Byte BlockHeader, where
+the first octet represents the EthashVersion enumeration value required for the 
+block. If this octet is not present, it may be assumed to be zero.
+  
+## Rationale
+
+This EIP is aimed at breaking existing ASIC based miners via small changes to the
+existing ethash algorithm.  We hope to accomplish the following:
+
+1. Break existing ASIC based miners.
+2. Demonstrate a willingness to fork in the event of future ASIC miner production.
+
+Goal #1 is something that we can only do probabalistically without detailed
+knowledge of existing ASIC miner design.  Our approach should balance the
+inherent security risks involved with changing the mining algorithm with the
+risk that the change we make does not break existing ASIC miners.  This EIP
+leans towards minimizing the security risks by making minimal changes to the
+algorithm accepting the risk that the change may not break dedicated hardware 
+miners that utilize partially or fully configurable logic. 
+
+Furthermore, we do not wish to introduce significant algorithm changes that
+may alter the power utilization or performance profile of existing GPU hardware.
+
+The change of FNV constant is a minimal change that can be quickly
+implemented across the various network node and miner implementations.
+
+It is proposed that `ASIC_MITIGATION_FORK_BLKNUM` be no more than 5550000 (epoch 185), giving
+around 30 days of notice to node and miner developers and a sufficient window
+for formal analysis of the changes by experts. We must weigh this window against
+the risk introduced by allowing ASICs to continue to probalistically propagate
+on the network, as well as the risk of providing too much advanced warning to 
```

Comment:
where did these requirements come from???

Reply:
Considering the change is fairly trivial and bitmain has been proven time and time again as a bad actor i see no reason to delay

Reply:
For the sake of argument, a cost analysis suggests bitmain could likely cross threshold to execute a 51% attack with about $300-400 MIllion in hardware. Analysts data suggests they do have that kind of capital available, and also  arguably the manufacturing capability if all resources had been devoted to such an effort. I also find such an attack extremely, but they are one of very few entities that could.  In these types of scenarios we have to consider every option where’s the risk of inaction, however small the chance, greatly outweighs the cost of action.

I don’t believe a 51% attack is necessary to significantly damage Ethereum either. A single player visibly gaining a large stake can serve to drive down mining interest through discouraging actors  as well as profitability below the level where most can afford to proceed based on power costs, leading to heavy geographic centralization. This may be inevitable on a long enough time scale, but the concern is POS isn’t ready to take the reigns yet.

Finally, there is an argument to be made that rapidly executing this fork allows us to gather definitive data on the current concentration of dedicated hardware on the network and better assess this risk. Delaying too long (assuming the change would happen at all) opens a new lower cost scenario risk. There could be enough time for such an entity to replace their ASIC hashpower with programmable or commodity hardware hashpower long enough to convince the world they don’t exist. I’m not much for conspiracy theories but we have to consider all possibilities. 

I also must make my personal stance clear. I am against hardforking for the directly evident current risks, but there are unknown risks. Thus I believe if we should hardfork the cost needs to be lower than the worst case of the unknown risks. This pull request meets that criteria.

Reply:
oh it's just common sense then?  i thought we needed data... well they have probably taken over 30%+ already.. maybe more.  waiting too long will allow them to take over more than 50%.. is that worth the risk?

Reply:
@AirSquirrels nailed it.

Reply:
@cryptomined if we are attempting to keep the discussion technical, Bitmain would have to have pre-produced 250,000+ E3 miners to take account for even 20% of the current difficulty, at costs of at least $100,000,000. If those gradually came online since December at even pacethey wouldn’t have even paid for their miners yet. The R&D alone certainly cost $10M + before mass production would even begin. I don’t think they have that level of production capacity while maintaining shipments of all their other miners. Infact, all their other miner shipments combined for that period barely reach that number.